tests/pytest/test_np_views.py

```diff
@@ -25,48 +28,105 @@ def check_leak(func, tol=1.e6):
         raise MemoryError("Function did not correctly clean up")
     else:
         print("Function %s: PASSED" % func.__name__)
+        return True
+
 
 def build_mat():
     mat = psi4.core.Matrix(mat_size, mat_size)
     return mat
 
+
 def build_view_mat():
     mat = psi4.core.Matrix(mat_size, mat_size)
     view = mat.np
     return mat, view
 
+
 def build_viewh_mat():
     mat = psi4.core.Matrix(mat_size, mat_size)
     view = mat.np
     return mat, view
 
+
 def build_view_set_mat():
     mat = psi4.core.Matrix(mat_size, mat_size)
     view = mat.np
     view[:] = 5
     return mat, view
 
+
 def build_arr_mat():
     mat = psi4.core.Matrix(mat_size, mat_size)
     view = np.asarray(mat)
     return mat, view
 
+
 def build_copy_mat():
     mat = psi4.core.Matrix(mat_size, mat_size)
     view = np.array(mat)
     return mat, view
 
+
+@using_memory_profiler
+def test_build_mat():
+    assert(check_leak(build_mat))
+
+
+@using_memory_profiler
+def test_build_view_mat():
+    assert(check_leak(build_view_mat))
+
+
+@using_memory_profiler
+def test_build_viewh_mat():
+    assert(check_leak(build_viewh_mat))
+
+
+@using_memory_profiler
+def test_build_view_set_mat():
+    assert(check_leak(build_view_set_mat))
+
+
+@using_memory_profiler
+def test_build_arr_mat():
+    assert(check_leak(build_arr_mat))
+
+
+@using_memory_profiler
+def test_build_copy_mat():
+    assert(check_leak(build_copy_mat))
+
+
+@using_memory_profiler
+def test_totals():
+    start = snapshot_memory()
+
+    check_leak(build_mat)
+    check_leak(build_view_mat)
+    check_leak(build_viewh_mat)
+    check_leak(build_view_set_mat)
+    check_leak(build_arr_mat)
+    check_leak(build_copy_mat)
+
+    # Double check totals
+    diff = abs(start - snapshot_memory())
+    if diff > 1.e6:
+        raise MemoryError("\nA function leaked %d bytes of memory!" % diff)
+    else:
+        print("\nNo leaks detected!")
+
+
 if __name__ == "__main__":
```

Comment:
You can just nuke the main now thats its properly wrapped.

Reply:
Nuke the `if __name__ == "__main__":`? Then you can't run it as a script.

Or nuke the `def test_totals()`? That seemed to have an extra "end minus beginning" check.

Reply:
Oh, I would say nuke the entire main. But if you want it as a script, just have it call the function instead of duplicating?